lavalink/nodemanager.py

```diff
@@ -100,9 +116,24 @@ def find_ideal_node(self, region: str = None):
         best_node = self.find_ideal_node(node.region)
 
         if not best_node:
-            # TODO: Move players to a queue, wait for a node to connect?
-            log.error('Unable to move players, no available nodes!')
+            for player in node.players:
+                self.player_queue.append(player)
+
+            await self._wait_for_best_node(node)
+            log.error('Unable to move players, no available nodes! Trying to find one to connect to.')
             return
 
         for player in node.players:
             await player.change_node(best_node)
+
+    async def _wait_for_best_node(self, original_node):
+        while True:
+            best_node = self.find_ideal_node()
+
+            if best_node:
+                for player in self.player_queue:
+                    await player.change_node(best_node)
+                self.player_queue.clear()
+                log.info('Connecting all players that were connected to NODE-{} to NODE-{}'.format(original_node.name, best_node.name))
+                return
+            # await asyncio.sleep(10) Not sure if I need this here or not
```

Comment:
This is absolutely necessary.

Reply:
Only if we keep this while loop, which as you mentioned in your above comment, will cause a deadlock so I'd prefer if this chunk of code is moved to `node_manager.py/_node_connected` event. That way it won't constantly be polling.